packages/taquito/src/contract/rpc-estimate-provider.ts

```diff
@@ -247,14 +298,97 @@ export class RPCEstimateProvider extends OperationEmitter implements EstimationP
+    const estimateProperties = await this.prepareEstimate({ operation: ops, source: pkh });
+    if (!estimationOptions.includeRevealOperation && isRevealNeeded) {
+      estimateProperties.shift();
+    }
+    return this.createEstimateObject(estimateProperties);
+  }
+
+  /**
+   *
+   * @description Estimate gasLimit, storageLimit and fees to reveal the current account
+   *
+   * @returns An estimation of gasLimit, storageLimit and fees for the operation or undefined if the account is already revealed
+   *
+   * @param Estimate
+   */
+  async reveal(params?: RevealParams) {
+    const pkh = await this.signer.publicKeyHash();
+    if (await this.isAccountRevealRequired(pkh)) {
+      const DEFAULT_PARAMS = await this.getAccountLimits(pkh);
+      const op = await createRevealOperation({
+        ...params, ...DEFAULT_PARAMS
+      },
+        pkh,
+        await this.signer.publicKey()
+      )
+      const estimateProperties = await this.prepareEstimate({ operation: op, source: pkh });
+      return this.createEstimateObject(estimateProperties);
+    }
+  }
+
+  private isRevealRequiredForOpType(op: RPCOperation[]) {
+    let opRequireReveal = false;
+    for (const operation of op) {
+      if (isOpRequireReveal(operation)) {
+        opRequireReveal = true;
+      }
+    }
+    return opRequireReveal;
+  };
+
+  private async isRevealOpNeeded(op: RPCOperation[], pkh: string) {
+    return (!await this.isAccountRevealRequired(pkh) || !this.isRevealRequiredForOpType(op)) ? false : true;
+  }
+
+  private async addRevealOp(op: RPCOperation[], pkh: string) {
+    op.unshift(
+      await createRevealOperation({
+        ...{ fee: DEFAULT_FEE.REVEAL, gasLimit: DEFAULT_GAS_LIMIT.REVEAL, storageLimit: DEFAULT_STORAGE_LIMIT.REVEAL }
+      },
+        pkh,
+        await this.signer.publicKey())
+    )
+    return op;
+  }
+
+  // Create an Estimate instance that include both reveal and operation fees
+  private createEstimateObject(estimateProperties: EstimateProperties[]) {
+    let milligasLimit = 0;
+    let storageLimit = 0;
+    let opSize = 0;
+    let minimalFeePerStorageByteMutez = 0;
+    let baseFeeMutez: number | undefined;
+
+    estimateProperties.forEach(estimate => {
+      milligasLimit += estimate.milligasLimit;
+      storageLimit += estimate.storageLimit;
+      opSize += estimate.opSize;
+      minimalFeePerStorageByteMutez = Math.max(estimate.minimalFeePerStorageByteMutez, minimalFeePerStorageByteMutez);
+      if (estimate.baseFeeMutez) {
+        baseFeeMutez = baseFeeMutez ? baseFeeMutez + estimate.baseFeeMutez : estimate.baseFeeMutez;
+      }
+    })
+    return new Estimate(milligasLimit, storageLimit, opSize, minimalFeePerStorageByteMutez, baseFeeMutez);
+  }
+
+  // Create separate instances of Estimate for each operation from the batch operation
+  private createEstimateObjectsArray(estimateProperties: EstimateProperties[]) {
+    const estimates: Estimate[] = [];
```

Comment:
Consider using `Array.map` https://developer.mozilla.org/fr/docs/Web/JavaScript/Reference/Global_Objects/Array/map

Reply:
I made use of Array.map, thank you for suggesting.